src/Capability/Stream.hs

```diff
@@ -86,15 +86,34 @@ instance HasState tag [a] m => HasStream tag a (StreamStack m) where
 -- | Accumulate streamed values in forward order in a difference list.
 newtype StreamDList m (a :: *) = StreamDList (m a)
   deriving (Functor, Applicative, Monad, MonadIO, PrimMonad)
-instance HasWriter tag (DList a) m => HasStream tag a (StreamDList m) where
-  yield_ _ = coerce @(a -> m ()) $ tell @tag . DList.singleton
+instance HasStream tag (DList a) m => HasStream tag a (StreamDList m) where
+  yield_ _ = coerce @(a -> m ()) $ yield @tag . DList.singleton
   {-# INLINE yield_ #-}
 
+-- | Accumulate streamed values with their own monoid.
+newtype StreamLog m (a :: *) = StreamLog (m a)
+  deriving (Functor, Applicative, Monad, MonadIO, PrimMonad)
```

Comment:
With this change, `WriterLog` needs an instance for `HasStream`, so the strategy becomes `StreamLog` and goes here. There is a `WriterLog` alias for back compat.